src/qililab/qprogram/qprogram.py

```diff
@@ -655,23 +710,42 @@ def measure_reset(
 
             Args:
                 bus (str): Identifier of the measurement bus.
-                waveform (IQPair): Waveform played during measurement.
-                weights (IQPair): Weights used for demodulation/integration.
+                waveform (IQPair | str): Waveform played during measurement.
+                weights (IQPair | str): Weights used for demodulation/integration.
                 control_bus (str): Identifier of the control/reset bus.
-                reset_pulse (IQPair): Pulse used for active reset.
+                reset_pulse (IQPair | str): Pulse used for active reset.
                 trigger_address (int, optional): Trigger address for synchronization. Defaults to 1.
                 save_adc (bool, optional): Whether to save ADC data. Defaults to False.
             """
-
-            operation = MeasureReset(
-                bus=bus,
-                waveform=waveform,
-                weights=weights,
-                control_bus=control_bus,
-                reset_pulse=reset_pulse,
-                trigger_address=trigger_address,
-                save_adc=save_adc,
+            operation: (
+                Measure
+                | MeasureResetCalibrated
             )
+            if isinstance(waveform, IQPair) and isinstance(weights, IQPair) and isinstance(reset_pulse, IQPair):
+                operation = MeasureReset(
+                    bus=bus,
+                    waveform=waveform,
+                    weights=weights,
+                    control_bus=control_bus,
+                    reset_pulse=reset_pulse,
+                    trigger_address=trigger_address,
+                    save_adc=save_adc,
+                )
+            elif isinstance(waveform, str) and isinstance(weights, str) and isinstance(reset_pulse, str):
+                operation = MeasureResetCalibrated(
+                    bus=bus,
+                    waveform=waveform,
+                    weights=weights,
+                    control_bus=control_bus,
+                    reset_pulse=reset_pulse,
+                    trigger_address=trigger_address,
+                    save_adc=save_adc,
+                )
+
+            #  Raise an error if a calibrated component has been used in conjunction with a non calibrated one
+            elif any(isinstance(component, str) for component in (waveform, weights, reset_pulse)):
+                raise NotImplementedError("For the waveform, weight, and reset pulse, you must either use the calibration file for all three or not use it at all.")
```

Comment:
Suggestion: add a log with a more meaningful message, like `"Missing calibration for reset_pulse"`